### ISSUE TITLE
jsonpath-maven-plugin is not displaying in maven-antrun-plugin version 1.8

### ISSUE THREAD

**Issue by shivajithanneeru** [ASzc/jsonpath-maven-plugin#3]
The property returned from jsonpath-maven-plugin is not displaying in maven-antrun-plugin version 1.8.  It works fine if the version of maven-antrun-plugin is 1.3

pom.xml file:
<project>
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.mycompany.app</groupId>
  <artifactId>my-module</artifactId>
  <version>1</version>
  <build>
    <plugins>
      <plugin>
        <groupId>ca.szc.maven</groupId>
        <artifactId>jsonpath-maven-plugin</artifactId>
        <version>1.0.0</version>
        <executions>
          <execution>
            <id>read-properties</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>set-properties</goal>
            </goals>
            <configuration>
              <file>package.json</file>
              <properties>
                <vsoVersion>$.subComponentDetails[0].upgradeVersion</vsoVersion>
                <componentName>$.subComponentDetails[0].componentName</componentName>
              </properties>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <version>1.8</version>
        <executions>
          <execution>
            <id>update-ppdm-package-json</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <target>
                       <echo>$${vsoVersion}</echo>
              </target>
            </configuration>
          </execution>  
        </executions>
      </plugin>
    </plugins>
  </build>

**Comment by shivajithanneeru** [comment]
The property returned from jsonpath-maven-plugin is not displaying in maven-antrun-plugin version 1.8. It works fine if the version of maven-antrun-plugin is 1.3

`<project>
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.mycompany.app</groupId>
  <artifactId>my-module</artifactId>
  <version>1</version>
  <build>
    <plugins>
      <plugin>
        <groupId>ca.szc.maven</groupId>
        <artifactId>jsonpath-maven-plugin</artifactId>
        <version>1.0.0</version>
        <executions>
          <execution>
            <id>read-properties</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>set-properties</goal>
            </goals>
            <configuration>
              <file>package.json</file>
              <properties>
                <vsoVersion>$.subComponentDetails[0].upgradeVersion</vsoVersion>
                <componentName>$.subComponentDetails[0].componentName</componentName>
              </properties>
            </configuration>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-antrun-plugin</artifactId>
        <version>1.8</version>
        <executions>
          <execution>
            <id>update-ppdm-package-json</id>
            <phase>prepare-package</phase>
            <goals>
              <goal>run</goal>
            </goals>
            <configuration>
              <target>
                      <echo>"${vsoVersion}"</echo>
              </target>
            </configuration>
          </execution>
        </executions>
      </plugin>
    </plugins>
  </build>
</project>`

**Comment by ASzc** [comment]
That is weird.

I notice that you're binding the executions of both plugins to the `prepare-package` phase. In theory maven is supposed to execute such executions in the order they're defined in the POM, but I'd recommend trying to use different phases.

`prepare-package` phase is unusually late for the `set-properties` goal. Try reading it in the phase the package.json is generated in, or if package.json is part of the sources, use the `initialize` phase.

**Comment by shivajithanneeru** [comment]
No luck even I used the jsonpath-maven-plugin in prepare-package and maven-antrun-plugin version in package phase. 
maven-antrun-plugin version 1.8 is unable to receive the properties from jsonpath-maven-plugin

**Comment by ASzc** [comment]
I determined that it is just the ant plugin doing this, as help:evaluate is ok with the property set by jsonpath-maven-plugin:set-properties.

I also determined that it is due to [a bug in maven-antrun-plugin](https://github.com/apache/maven-antrun-plugin/blob/f7c74861233953311d2926f906fe37c5a06f08f8/src/main/java/org/apache/maven/plugins/antrun/AntRunMojo.java#L421). Since 1.8, it only passes along properties that are in `mavenProject.getProperties()`, which does not include properties set by other maven plugins, as [jsonpath-maven-plugin does here](https://github.com/ASzc/jsonpath-maven-plugin/blob/0523d25823f88de4ef1e4f58019926f4ddc78bcd/src/main/java/ca/szc/maven/jsonpath/SetPropertiesMojo.java#L60) in `session.getUserProperties()`.

[MANTRUN-178](https://issues.apache.org/jira/browse/MANTRUN-178) introduced this bug. I will file an issue and try to make a pull request to fix this bug.

The only workaround I think is to use a version less than 1.8 for the time being. I tried forcing the property in there with a property task, but it doesn't work.

Here's my MWE:

pom.xml:
```
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.mycompany.app</groupId>
    <artifactId>my-module</artifactId>
    <version>1</version>
    <properties>
        <version.static>1.2.3</version.static>
    </properties>
    <build>
        <plugins>
            <plugin>
                <groupId>ca.szc.maven</groupId>
                <artifactId>jsonpath-maven-plugin</artifactId>
                <version>1.0.0</version>
                <executions>
                    <execution>
                        <id>read-properties</id>
                        <phase>initialize</phase>
                        <goals>
                            <goal>set-properties</goal>
                        </goals>
                        <configuration>
                            <file>package.json</file>
                            <properties>
                                <vsoVersion>$.version</vsoVersion>
                            </properties>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-help-plugin</artifactId>
                <version>3.2.0</version>
                <executions>
                    <execution>
                        <id>echo</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>evaluate</goal>
                        </goals>
                        <configuration>
                            <expression>vsoVersion</expression>
                        </configuration>
                    </execution>
                    <execution>
                        <id>echo-static</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>evaluate</goal>
                        </goals>
                        <configuration>
                            <expression>version.static</expression>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-antrun-plugin</artifactId>
                <version>1.8</version>
                <executions>
                    <execution>
                        <id>update-ppdm-package-json</id>
                        <phase>prepare-package</phase>
                        <goals>
                            <goal>run</goal>
                        </goals>
                        <configuration>
                            <target>
                                <echo>"${vsoVersion}"</echo>
                                <echo>"${version.static}"</echo>
                            </target>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
</project>
```

package.json:
```
{
    "version": "3.2.1"
}
```

**Comment by ASzc** [comment]
PR filed: apache/maven-antrun-plugin#1

**Comment by shivajithanneeru** [comment]
Thanks

**Comment by shivajithanneeru** [comment]
I approved your PR on maven-antrun-plugin

**Comment by ASzc** [comment]
Another workaround might be to specify the property with any value in the pom. This would put it in the list that antrun will iterate through when it copies. Unsure which value will take precedence.

```
<properties>
    <vsoVersion></vsoVersion>
</properties>
```

**Comment by shivajithanneeru** [comment]
Nope. It doesn't work.
`
[INFO] --- maven-antrun-plugin:1.3:run (Display) @ my-module ---
[INFO] Executing tasks
     [echo] ""
[INFO] Executed tasks
`

**Comment by shivajithanneeru** [comment]
Hi @ASzc,
I appreciate all your help in figuring out the bug in maven-antrun-plugin and creating PR for it.
Can you please let me know, when will the PR got merged to master?

**Comment by ASzc** [comment]
I can update you when it happens, sure. It might be more reliable to press subscribe on the PR comment thread though.

I can't predict when it will be merged unfortunately. I don't think a speedy turnaround is likely, based on the reaction I've seen so far, and the fact maven-antrun-plugin isn't very actively developed.

**Comment by shivajithanneeru** [comment]
Hi @ASzc , Any update on the PR you submitted?

**Comment by ASzc** [comment]
I pinged the PMC chair, will see if that creates any movement.

**Comment by shivajithanneeru** [comment]
@ASzc Any update?

**Comment by ASzc** [comment]
Fix was merged. It'll be in the next release, but there doesn't seem to be one planned immediately.

**Comment by shivajithanneeru** [comment]
@ASzc Thanks

**Comment by ASzc** [comment]
The fix is included as part of [maven-antrun-plugin 3.0.0](https://github.com/apache/maven-antrun-plugin/blob/maven-antrun-plugin-3.0.0/src/main/java/org/apache/maven/plugins/antrun/AntRunMojo.java#L423-L425).

Closing this issue.
